yt/frontends/gadget/data_structures.py

```diff
@@ -559,8 +559,12 @@ def _is_valid(cls, filename, *args, **kwargs):
         return header.validate()
 
 
+class GadgetHDF5File(ParticleFile):
+    pass
+
+
 class GadgetHDF5Dataset(GadgetDataset):
-    _file_class = ParticleFile
+    _file_class = GadgetHDF5File
```

Comment:
I don't immediately get why this new hollow class is necessary

Reply:
That's because this was supposed to be largely no-op!  But upcoming changes will connect this with Swift.

Additionally, we want `ParticleFile` to be an ABC.

Reply:
fair enough